src/mon/OSDMonitor.cc

```diff
@@ -606,6 +725,43 @@ void OSDMonitor::update_from_paxos(bool *need_bootstrap)
   }
 }
 
+void OSDMonitor::register_cache_with_pcm()
+{
+  if (mon_memory_target <= 0 || mon_memory_min <= 0) {
+    derr << __func__ << " Invalid memory size specified for mon caches."
+         << " Caches will not be auto-tuned."
+         << dendl;
+    return;
+  }
+  uint64_t base = mon_memory_base;
+  double fragmentation = mon_memory_fragmentation;
+  // For calculating total target memory, consider rocksdb cache size.
+  uint64_t target = mon_memory_target + rocksdb_cache_size;
```

Comment:
the idea with the <daemon>_memory_target options is to let the user/kubernetes/etc. provide an amount of bytes for the whole process to stick to - that's why in bluestore this target is just set to osd_memory_target. Here it should be ```target = mon_memory_target```, and the monitor should adjust whatever internals it can to try to stay within that bound (so far inc map, full map, and rocksdb cache sizes)

Reply:
With the current code, rocksdb's cache size will not change when rocksdb_cache_size is set at runtime. To change the size of the rocksdb cache, you'd need to call set_ratio() on rocksdb's binned lru cache, and adjust the inc + full cache ratios correspondingly.

e.g. divide the rocksdb cache size into the available bytes to get the kv cache ratio, and split the remainder between the inc and full caches.

Reply:
The ratios are calculated internally as per your suggestion. Please verify.